diff -uNr a/rgmanager/src/resources/fs.sh.in b/rgmanager/src/resources/fs.sh.in
--- a/rgmanager/src/resources/fs.sh.in	2015-11-09 13:37:37.366125692 +0100
+++ b/rgmanager/src/resources/fs.sh.in	2015-11-09 13:41:02.503088930 +0100
@@ -240,7 +240,7 @@
 	[ -z "$OCF_RESKEY_fstype" ] && return 0
 
 	case $OCF_RESKEY_fstype in
-	ext2|ext3|ext4|btrfs|jfs|xfs|reiserfs|vfat|tmpfs|vxfs)
+	ext2|ext3|ext4|btrfs|jfs|xfs|reiserfs|vfat|vxfs)
 		return 0
 		;;
 	*)
@@ -357,13 +357,6 @@
 			esac
 			;;
 
-		tmpfs)
-			case $o in
-			size=*|nr_blocks=*|mode=*)
-				continue
-				;;
-			esac
-			;;
 		btrfs)
 			# tbd
 			continue
diff -uNr a/rgmanager/src/resources/utils/fs-lib.sh b/rgmanager/src/resources/utils/fs-lib.sh
--- a/rgmanager/src/resources/utils/fs-lib.sh	2015-11-09 13:37:37.373125591 +0100
+++ b/rgmanager/src/resources/utils/fs-lib.sh	2015-11-09 13:32:28.000000000 +0100
@@ -782,6 +782,19 @@
 	declare mount_options=""
 
 	#
+	# Check if fstype is supported
+	#
+	verify_fstype
+	case $? in
+	$OCF_ERR_ARGS)
+		ocf_log err "File system type $OCF_RESKEY_fstype not supported"
+                return $OCF_ERR_ARGS
+		;;
+	*)
+		;;
+	esac
+
+	#
 	# Check if mount point was specified.  If not, no need to continue.
 	#
 	case "$mp" in
