From 3c72afb4feba7a33c92329f25e90f35bf2797f7e Mon Sep 17 00:00:00 2001
From: Oyvind Albrigtsen <oalbrigt@redhat.com>
Date: Mon, 14 Dec 2015 13:08:17 +0100
Subject: [PATCH] rgmanager: netfs move defaults to metadata

---
 rgmanager/src/resources/netfs.sh | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/rgmanager/src/resources/netfs.sh b/rgmanager/src/resources/netfs.sh
index 5dad993..dc64cd7 100755
--- a/rgmanager/src/resources/netfs.sh
+++ b/rgmanager/src/resources/netfs.sh
@@ -80,14 +80,14 @@ do_metadata()
 	    <content type="string"/>
         </parameter>
 
-        <parameter name="fstype" required="0">
+        <parameter name="fstype">
 	    <longdesc lang="en">
 	    	File System type (nfs, nfs4 or cifs)
 	    </longdesc>
             <shortdesc lang="en">
 	    	File System Type
             </shortdesc>
-	    <content type="string"/>
+	    <content type="string" default="nfs"/>
         </parameter>
 
         <parameter name="no_unmount" required="0">
@@ -464,11 +464,6 @@ do_force_unmount() {
 
 populate_defaults()
 {
-	if [ -z "$OCF_RESKEY_fstype" ]; then
-		export OCF_RESKEY_fstype=nfs
-	fi
-
-
         case $OCF_RESKEY_fstype in
 	nfs|nfs4)
 		export OCF_RESKEY_device="$OCF_RESKEY_host:$OCF_RESKEY_export"
