From fc497e888afcb88babbc21a59883556335c070fa Mon Sep 17 00:00:00 2001
From: Oyvind Albrigtsen <oalbrigt@redhat.com>
Date: Fri, 31 Aug 2018 11:41:32 +0200
Subject: [PATCH] aliyun-vpc-move-ip: improve metadata and manpage

---
 heartbeat/aliyun-vpc-move-ip | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/heartbeat/aliyun-vpc-move-ip b/heartbeat/aliyun-vpc-move-ip
index e27952adb..c004d26fc 100755
--- a/heartbeat/aliyun-vpc-move-ip
+++ b/heartbeat/aliyun-vpc-move-ip
@@ -123,7 +123,7 @@ ecs_ip_metadata() {
 Resource Agent to move IP addresses within a VPC of the Aliyun Webservices ECS
 by changing an entry in an specific routing table
 </longdesc>
-<shortdesc lang="en">Move IP within a APC of the Aliyun ECS</shortdesc>
+<shortdesc lang="en">Move IP within a VPC of the Aliyun ECS</shortdesc>
 <parameters>
 <parameter name="address" required="1">
 <longdesc lang="en">
@@ -134,21 +134,23 @@ VPC private IP address
 </parameter>
 <parameter name="routing_table" required="1">
 <longdesc lang="en">
-Name of the routing table, where the route for the IP address should be changed, i.e. rtb-...
+Name of the routing table, where the route for the IP address should be changed, i.e. vtb-...
 </longdesc>
 <shortdesc lang="en">routing table name</shortdesc>
 <content type="string" default="" />
 </parameter>
 <parameter name="interface" required="1">
 <longdesc lang="en">
-Name of the network interfacen, i.e. eth0
+Name of the network interface, i.e. eth0
 </longdesc>
 <shortdesc lang="en">network interface name</shortdesc>
 <content type="string" default="eth0" />
 </parameter>
 <parameter name="profile" required="0">
 <longdesc lang="en">
-Valid Aliyun CLI profile name
+Valid Aliyun CLI profile name (see 'aliyuncli-ra configure').
+
+See https://www.alibabacloud.com/help/doc-detail/43039.htm?spm=a2c63.p38356.b99.16.38a914abRZtOU3 for more information about aliyuncli-ra.
 </longdesc>
 <shortdesc lang="en">profile name</shortdesc>
 <content type="string" default="default" />
