From 059effc058758c1294d80f03741bf5c078f1498d Mon Sep 17 00:00:00 2001
From: Oyvind Albrigtsen <oalbrigt@redhat.com>
Date: Fri, 13 Jul 2018 13:22:56 +0200
Subject: [PATCH 2/3] CI: skip Python agents in shellcheck

---
 ci/build.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ci/build.sh b/ci/build.sh
index 608387ad..c331e9ab 100755
--- a/ci/build.sh
+++ b/ci/build.sh
@@ -58,6 +58,7 @@ check_all_executables() {
 	echo "Checking executables and .sh files..."
 	while read -r script; do
 		file --mime "$script" | grep 'charset=binary' >/dev/null 2>&1 && continue
+		file --mime "$script" | grep 'text/x-python' >/dev/null 2>&1 && continue
 		head=$(head -n1 "$script")
 		[[ "$head" =~ .*ruby.* ]] && continue
 		[[ "$head" =~ .*zsh.* ]] && continue
-- 
2.17.1

