--- a/heartbeat/SAPInstance	2019-02-20 12:42:55.655819263 +0100
+++ b/heartbeat/SAPInstance	2019-02-08 10:57:02.281048136 +0100
@@ -159,14 +159,14 @@
   <content type="string" default="disp+work|msg_server|enserver|enrepserver|jcontrol|jstart|enq_server|enq_replicator"/>
  </parameter>
   <parameter name="SHUTDOWN_METHOD" unique="0" required="0">
-  <longdesc lang="en">Usual a SAP Instance is stopped by the command 'sapcontrol -nr InstanceNr -function Stop'. SHUTDOWN_METHOD=KILL means to kill the SAP Instance using OS commands. SAP processes of the instance are terminated with 'kill -9', shared memory is deleted with 'cleanipc' and the 'kill.sap' file will be deleted. That method is much faster than the gracefull stop, but the instance does not have the chance to say goodbye to other SAPinstances in the same system. USE AT YOUR OWN RISK !!</longdesc>
+  <longdesc lang="en">Usually a SAP Instance is stopped by the command 'sapcontrol -nr InstanceNr -function Stop'. SHUTDOWN_METHOD=KILL means to kill the SAP Instance using OS commands. SAP processes of the instance are terminated with 'kill -9', shared memory is deleted with 'cleanipc' and the 'kill.sap' file will be deleted. That method is much faster than the graceful stop, but the instance does not have the chance to say goodbye to other SAPinstances in the same system. USE AT YOUR OWN RISK !!</longdesc>
   <shortdesc lang="en">Shutdown graceful or kill a SAP instance by terminating the processes. (normal|KILL)</shortdesc>
   <content type="string" default="normal"/>
  </parameter>
  <parameter name="ERS_InstanceName" unique="1" required="0">
   <longdesc lang="en">Only used in a Master/Slave resource configuration:
 The full qualified SAP enqueue replication instance name. e.g. P01_ERS02_sapp01ers. Usually this is the name of the SAP instance profile.
-The enqueue replication instance must be installed, before you want to configure a master-slave cluster recource.
+The enqueue replication instance must be installed, before you want to configure a master-slave cluster resource.
 
 The master-slave configuration in the cluster must use this properties:
 clone_max = 2
@@ -209,7 +209,7 @@
   <longdesc lang="en">Only used for ASCS/ERS SAP Netweaver installations without implementing a master/slave resource to
     allow the ASCS to 'find' the ERS running on another cluster node after a resource failure. This parameter should be set
     to true 'only' for the ERS instance for implementations following the SAP NetWeaver 7.40 HA certification (NW-HA-CLU-740). This includes also
-    systems for NetWeaver less than 7.40, if you like to impelemnt the NW-HA-CLU-740 scenario.
+    systems for NetWeaver less than 7.40, if you like to implement the NW-HA-CLU-740 scenario.
   </longdesc>
   <shortdesc lang="en">Mark SAPInstance as ERS instance</shortdesc>
   <content type="boolean" default="false" />
@@ -225,7 +225,7 @@
 <action name="monitor" depth="0" timeout="60s" interval="119s" role="Master" />
 <action name="promote" timeout="320s" />
 <action name="demote" timeout="320s" />
-<action name="reload" timeout="320" />
+<action name="reload" timeout="320s" />
 <action name="validate-all" timeout="5s" />
 <action name="meta-data" timeout="5s" />
 <action name="methods" timeout="5s" />
